test/language-transformer-cycles.test.js

```diff
@@ -104,39 +100,36 @@ function arraysAreEqual(rules1, rules2) {
     return true;
 }
 
-const dirname = pathDirname(fileURLToPath(import.meta.url));
-const descriptors = [];
-const languageSummaries = getLanguageSummaries();
-for (const {languageTransformsFile} of languageSummaries) {
-    if (!languageTransformsFile) { continue; }
-    /** @type {import('language-transformer').LanguageTransformDescriptor} */
-    const descriptor = parseJson(readFileSync(join(dirname, `../ext/${languageTransformsFile}`), {encoding: 'utf8'}));
-    descriptors.push(descriptor);
-}
+const languagesWithTransforms = getAllLanguageTransformDescriptors();
 
-describe.each(descriptors)('Cycles Test $language', (descriptor) => {
+describe.each(languagesWithTransforms)('Cycles Test $iso', ({languageTransforms}) => {
     test('Check for cycles', ({expect}) => {
         const languageTransformer = new LanguageTransformer();
-        languageTransformer.addDescriptor(descriptor);
+        languageTransformer.addDescriptor(languageTransforms);
 
         /** @type {RuleNode[]} */
         const ruleNodes = [];
-        for (const [groupName, reasonInfo] of Object.entries(descriptor.transforms)) {
+        for (const [groupName, reasonInfo] of Object.entries(languageTransforms.transforms)) {
             for (const rule of reasonInfo.rules) {
-                ruleNodes.push(new RuleNode(groupName, rule));
+                if (rule.type === 'suffix') {
```

Comment:
I patched up the cycles test so that it still works on suffix rules. Passing the cycles test now doesn't mean there are no cycles, but failing it means there are. 

I'm pretty sure it can also be made to cover prefixes, but I don't think a general cycles test is feasible.